src/Entity/Form/SubscriptionForm.php

```diff
@@ -125,7 +169,97 @@ public function save(array $form, FormStateInterface $form_state) {
       }
     }
     catch (\Exception $e) {
-      $this->messenger->addError($e->getMessage());
+      $previous = $e->getPrevious();
+
+      // If insufficient funds error, format nicely and add link to add credit.
+      if ($previous instanceof ClientErrorException && $previous->getEdgeErrorCode() === static::INSUFFICIENT_FUNDS_ERROR) {
+        preg_match_all('/\[(?\'amount\'.+)\]/', $e->getMessage(), $matches);
```

Comment:
This looks like something the client php library should do, handle and extract errors into clean objects along with tests to validate the errors clients depend on are not breaking.